### PR TITLE
Adding support for speclite and general broadband filters 

### PR DESCRIPTION
**Pull request by Henry-Best-01**
Allows calculation of agn response function using speclite filter Allows user defined filters to be created and used with agn response functions Allows conversion of any filters to source plane where reprocessing must take place Allows unit conversion between any astropy units
Added a plot in jupyter notebook showing a simple usage of these functions

### CODE REVIEW

**Comment by nkhadka21** [comment]
Hi @Henry-Best-01 , thank you very much for this PR! What you have done is great. Did this PR gives quasar light curve in a specific band directly from variability class? If not, it would be great if you could connect all your development to variability class (mainly passing light curve in specified band to variability class instead of intrisic signal). Thank you!

**Review by nkhadka21 — COMMENTED** [review]
Hi @Henry-Best-01 , thank you very much for this PR! What you have done is great. Did this PR gives quasar light curve in a specific band directly from variability class? If not, it would be great if you could connect all your development to variability class (mainly passing light curve in specified band to variability class instead of intrisic signal). Thank you!

**Comment by sibirrer** [comment]
@ajshajib it looks like none of the PRs execute a coverage report. Is there something broken there or did we reach a limit for the free version? Thank you!

**Comment by ajshajib** [comment]
@Henry-Best-01, can you kindly pull and merge from the main branch? After that, the coverage report will start working for this PR.

**Review by nkhadka21 — CHANGES_REQUESTED** [review]
Hi @Henry-Best-01 , thank you very much for this PR! it seems like this work is almost at the end. However, I have some suggestions for managing the code. Currently Variability class looks very crowded with mix the mixture of many things. So, I have suggested to break the code in different part and put them in different functions outside of the Variability class and call those functions whenever needed.

File: slsim/Sources/SourceVariability/variability.py [hunk 1]
```diff
@@ -64,3 +119,130 @@ def variability_at_time(self, observation_times):
         :return: variability at given time.
         """
         return self._model(observation_times)
+
+    def parse_kwargs_for_lamppost_reprocessed_model(self):
```
Comment (nkhadka21) [inline slsim/Sources/SourceVariability/variability.py:123]: This function is still inside the Variability class. Is it possible to move this function outside of variability class? If it is necessary to put this function inside the variability class, I am ok with this.

File: slsim/Sources/SourceVariability/variability.py [hunk 2]
```diff
@@ -64,3 +119,130 @@ def variability_at_time(self, observation_times):
...
+                self.driving_signal_kwargs[kwarg] = self.kwargs_model[kwarg]
+
+            elif kwarg in ["redshift"]:
+                self.redshift = self.kwargs_model[kwarg]
+            elif kwarg in ["delta_wavelength"]:
+                self.delta_wavelength = self.kwargs_model[kwarg]
+            else:
+                self.reprocessing_kwargs[kwarg] = self.kwargs_model[kwarg]
+
+    def reprocess_with_lamppost_model(self):
```
Comment (nkhadka21) [inline slsim/Sources/SourceVariability/variability.py:181]: This function is also inside the variability class. Is it possible to move this function outside of the variability class. If it is necessary to put this function inside the variability class, I am ok with this.

**Review by nkhadka21 — COMMENTED** [review]
@Henry-Best-01 , thank you very much for this update! We are very close to the completion of this PR. I have still some minor comments. Please look at them.

**Comment by nkhadka21** [comment]
Hi @Henry-Best-01, I noticed a conflict here. I hope this is not very difficult to resolve in your side.

**Comment by Henry-Best-01** [comment]
Sorry for the delay! It's been very busy for me lately. I will try to resolve the conflict and set this up as we discussed last month.

**Comment by nkhadka21** [comment]
Hi @Henry-Best-01 , the CI/build  is failing because of the very small skyarea used in that particular test. I have updated this. Please synchronize your branch with main and it should work. Thank you!

**Review by nkhadka21 — APPROVED** [review]
Hi @Henry-Best-01 , thank you very much for these updates! Looks very good. I approve this PR.

**Pull request merged**